vm/class.go

```diff
@@ -371,6 +371,74 @@ func builtinCommonInstanceMethods() []*BuiltInMethodObject {
 				}
 			},
 		},
+		{
+			// Returns true if Object class is equal to the input argument class
+			//
+			// ```ruby
+			// "Hello".is_a(String)            # => true
+			// 123.is_a(Integer)               # => true
+			// [1, true, "String"].is_a(Array) # => true
+			// { a: 1, b: 2 }.is_a(Hash)       # => true
+			// "Hello".is_a(Integer)           # => false
+			// 123.is_a(Range)                 # => false
+			// (2..4).is_a(Hash)               # => false
+			// nil.is_a(Integer)               # => false
+			// ```
+			//
+			// @param n/a []
+			// @return [Boolean]
+			Name: "is_a",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+					if len(args) != 1 {
+						return t.vm.initErrorObject(ArgumentError, "Expect 1 argument. got: %d", len(args))
+					}
+
+					c := args[0]
+					gobyClass, ok := c.(*RClass)
+
+					if !ok {
+						return t.vm.initErrorObject(TypeError, WrongArgumentTypeFormat, classClass, c.Class().Name)
+					}
+
+					receiverClass := receiver.Class()
+
+					for {
```

Comment:
I like the way you implement this, quite neat. Maybe you can try writing recursive anonymous function next time 👍